Select: Add ItemDescription component to show extra data in options

ariser opened the pull request:
## Feature

Adding a component that allows this:

![image](https://github.com/user-attachments/assets/c2e7d899-93bb-4517-b7bc-cfa7357b56cf)

crisalbu commented:
Thank you for addressing this @ariser !!!! ❤️ 

Could we have the prop be: `Description` and values `false / true`?

ariser commented:
Yeeaa, this is too confusing.
This control you're seeing in storybook isn't a part of the component API. It's there specifically to control how the componsne is rendered in storybook. Which is in no way how it should be done.
In real use you simply either specify the `options` prop, or define options via children

I'll redo stories for Selects as well, will make them clearer and more useful.

ariser commented:
- updated how children are processed in the description component
- fixed stories for all Selects
  - all props are now inferred from the component type
  - all prop types are described in stories
  - different ways to build options are presented in different stories
- supported `description` field in `options`

ariser commented:
> For disabled items, could we make the description text darker in dark mode / lighter in light mode. Maybe use the disabled text color?

it _is_ the disabled text color ¯\\\_(ツ)_/¯ 

in dark theme, the disabled item has bright background for some reson — this is incorrect. This is an issue with dark theme palette, we need to look into that

by the way, having a “disabled” color is not the best approach. A disabled color varies depending on where an item is displayed. And as a rule, “disabled” is not a color, disabled items have low contrast. So it's easier and more convenient to use opacity rather than color.

ariser commented on src/components/Select/common/InternalSelect.tsx:
```diff
@@ -422,24 +425,20 @@ export const InternalSelect = ({
                                     return (
                                       <MultiSelectCheckboxItem
                                         key={`select-${id}-group-${index}-item-${itemIndex}`}
                                         {...itemProps}
```
there are a lot of props that are passed into this component here, not sure why it makes sense to treat just the label differently?

crisalbu commented:
> So it's easier and more convenient to use opacity rather than color.

I agree, this would work better on different backgrounds too. Could you open an issue in Click-UI for this?

crisalbu reviewed (approved):
Yep, all good, thanks for making these changes!
